nemo/collections/vision/data/megatron/image_folder.py

```diff
@@ -0,0 +1,286 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# code taken from
+# https://github.com/pytorch/vision/blob/main/torchvision/datasets/folder.py
+# added support for classes_fraction and data_per_class_fraction
```

Comment:
Do we need to take code from here?

Reply:
This is actually from megatron code here: 
https://github.com/NVIDIA/Megatron-LM/blob/main/megatron/data/image_folder.py

Do we want to just import it from megatron directly since we have mcore dependency now